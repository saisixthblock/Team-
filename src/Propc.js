import React from "react";

const Propc = (props) => {
  return <div>{props.count}</div>;
};

export default Propc;
