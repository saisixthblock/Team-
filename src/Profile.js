import React, { useState } from "react";
import Propc from "./Propc";
const Profile = () => {
  const [count, setCount] = useState(5);
  const [name, setName] = useState("sixth");
  return (
    <div>
      <h1>count:{count}</h1>
      <p>sucessfully counted{name}</p>

      <button onClick={() => setCount(count + 5)}>click me</button>
      <Propc counts={count} />
    </div>
  );
};

export default Profile;
