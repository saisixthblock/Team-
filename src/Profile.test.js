import React from "react";
import Profile from "./Profile";
import Propc from "./Propc";
import { shallow } from "enzyme/build";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
configure({ adapter: new Adapter() });
test("checking h1 tag", () => {
  const wrapper = shallow(<Profile />);
  expect(wrapper.find("h1").text()).toContain("count:5");
});
test("checking button tag", () => {
  const wrapper = shallow(<Profile />);
  wrapper.find("button").simulate("click");
  expect(wrapper.find("h1").text()).toBe("count:10");
});
test("checking props", () => {
  const wrapper = shallow(<Profile />);
  expect(wrapper.find(Propc).prop("counts")).toBe(5);
});
test("checking props1", () => {
  const tree = shallow(<Profile />);
  expect(toJson(tree)).toMatchSnapshot();
});
