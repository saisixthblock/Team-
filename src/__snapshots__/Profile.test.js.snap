// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking props1 1`] = `
<div>
  <h1>
    count:
    5
  </h1>
  <p>
    sucessfully counted
    sixth
  </p>
  <button
    onClick={[Function]}
  >
    click me
  </button>
  <Propc
    counts={5}
  />
</div>
`;
